cabal-install/Distribution/Client/SetupWrapper.hs

```diff
@@ -334,7 +335,8 @@ selfExecSetupMethod verbosity options _pkg bt mkargs = do
 
   searchpath <- programSearchPathAsPATHVar
                 (getProgramSearchPath (useProgramConfig options))
-  env        <- getEffectiveEnvironment [("PATH", Just searchpath)]
```

Comment:
@23Skidoo I understand not—I'd have expected the outer `cabal $CMD` (say `install`) to override `HASKELL_DIST_DIR` (as it does); the change should be behavior-preserving, other than the extra variable in the environment.